back-end/app/Http/Controllers/IngredientsController.php

```diff
@@ -1,30 +1,84 @@
-<?php
+<?php declare(strict_types=1);
 
 namespace App\Http\Controllers;
 
+use App\Http\Requests\Ingredient as IngredientReq;
+use App\Repositories\Implementations\Ingredient as IngredientImp;
+use Illuminate\Http\JsonResponse;
+
 class IngredientsController extends Controller
 {
+    /** @var IngredientImp $ingredientImp */
+    private $ingredientImp;
+
+    public function __construct(IngredientImp $ingredientImp)
+    {
+        $this->ingredientImp = $ingredientImp;
+    }
+
     /**
-     * Display a listing of the resource.
+     * Get all Ingredient from DB.
      *
-     * @return \Illuminate\Http\Response
+     * @return JsonResponse
      */
-    public function index()
+    public function index(): JsonResponse
     {
-        return response()->json(['code' =>"Tutti gli ingredienti"],200);
+        $data = $this->ingredientImp->getAll();
 
+        return response()->json(['data' => $data], self::OK);
     }
 
     /**
-     * Display the specified resource.
+     * Get the specified Ingredient from DB.
      *
      * @param  int  $id
-     * @return \Illuminate\Http\Response
+     * @return JsonResponse
+     */
+    public function show(int $id): JsonResponse
+    {
+        $data = $this->ingredientImp->getById($id);
+
+        return response()->json(['data' => $data], self::OK);
```

Comment:
Non mi ricordo, la proprietà `data` è quella di default della risposta o un'aggiunta? Se aggiunta andrei a restituire direttamente l'oggetto trovato.

Non vedo gestita l'eccezione se non viene trovato. Usiamo quindi l'eccezioni di Laravel o ci mappiamo i nostri errori?
Comunque si potrebbe farlo in un secondo momento nel caso. Giusto per segnarsi dei task da fare.